src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -69,7 +73,11 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	defer connOut.Close()
+	defer func() {
+		if err := connOut.Close(); err != nil {
+			log.Debugf("Error closing the out connection: %s", err)
+		}
+	}()
```

Comment:
OK, what about skipping both of these deferred close calls and instead defining your closeConns call here and both passing it to pipeData and deferring it here? That would eliminate a tad bit of duplicate code on the closes and I think would accomplish the same thing.